hypervisor/virt/rhevm/rhevmcli.py

```diff
@@ -88,3 +88,98 @@ def info(self):
+        :return: the value you want to get
+        """
+        cmd = f"ovirt-shell -c -E 'show {object_type} {object_id}' |grep '^{value}'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and value in output:
+            result = output.strip().split(':')[1].strip()
+            logger.info(f"Succeeded to get rhevm {object_type} ({object_id}) {value}: {result}")
+            return result
+        else:
+            logger.info(f"Failed to get rhevm {object_type} ({object_id}) {value}")
+
+    def get_guest_ip_by_mac(self, guest_name, host_ip, host_user, host_pwd):
+        """
+        Get guest ip by mac
+        :param guest_name: name for the specific guest
+        :param host_ip: the ip for the RHEVM host
+        :param host_user: the ssh username for the RHEVM host
+        :param host_pwd: the ssh password for the RHEVM host
+        :return:
+        """
+        gateway = self.get_gateway(host_ip, host_user, host_pwd)
+        guest_mac = self.get_guest_mac(guest_name)
+        option = "grep 'Nmap scan report for' | grep -Eo '([0-9]{1,3}[\.]){3}[0-9]{1,3}'| tail -1"
+        cmd = f"nmap -sP -n {gateway} | grep -i -B 2 {guest_mac} | {option}"
+        ret, output = SSHConnect(host_ip, host_user, host_pwd).runcmd(cmd)
+        if not ret and output is not None and output is not "":
+            guest_ip = output.strip()
+            logger.info(f"Succeeded to get rhevm guest ip ({guest_ip})")
+            return output.strip()
+        else:
+            logger.info(f"Failed to get rhevm guest ip")
+
+    def get_gateway(self, host_ip, host_user, host_pwd):
+        """
+        Get the gateway by ip route command
+        :param host_ip: the ip for the RHEVM host
+        :param host_user: the ssh username for the RHEVM host
+        :param host_pwd: the ssh password for the RHEVM host
+        :return: the gateway for host
+        """
+        cmd = f"ip route | grep {host_ip}"
+        ret, output = SSHConnect(host_ip, host_user, host_pwd).runcmd(cmd)
+        if not ret and output is not None and output is not "":
+            output = output.strip().split(" ")
+            if len(output) > 0:
+                gateway = output[0]
+                return gateway
+        raise FailException(f"Failed to get gateway({host_ip})")
+
+    def get_guest_mac(self, guest_name):
+        """
+        Get the mac address for the guest
+        :param guest_name: name for the specific guest
+        :return: the mac address for the guest
+        """
+        cmd = f"ovirt-shell -c -E 'list nics --parent-vm-name {guest_name} --show-all' | grep  '^mac-address'"
+        ret, output = self.ssh.runcmd(cmd)
+        if not ret and "mac-address" in output:
+            mac_addr = output.strip().split(': ')[1].strip()
+            logger.info(f"rhevm({self.server}) guest mac is: {mac_addr}")
+            return mac_addr
+        else:
+            logger.info(f"Failed to check rhevm({self.server}) guest mac")
```

Comment:
Here if need to return a null value? I guess if the "cmd = f"nmap -sP -n {gateway} | grep -i -B 2 {guest_mac} | {option}" in def get_guest_ip_by_mac will fail with a None {guest_mac}...